src/jmatrix.jl

```diff
@@ -1,110 +1,108 @@
 
-struct JMatrix{T, N<:Integer, SGN} <: AbstractMatrix{T}
-    n::N    #size of the square matrix
-    sgn::SGN #+-1, allows to transpose,invert easily the matrix.
-    function JMatrix{T, N, SGN}(n, sgn) where {T, N<:Integer, SGN}
-        (sgn == T(1) || sgn == T(-1) ) || throw("sgn argument must be +-1")
-        new{T, N, SGN}(n,sgn)
-    end
-end
-"""
-    JMatrix(T, n, sgn)
 
-Creates an `AbstractMatrix{T}` of size n x n. The JMatrix 
-is a Block-diagonal matrix whose diagonal blocks are [0 1; -1 0].
-If n is odd, then the last block is the 1 x 1 zero block.
-sgn is +-1, set to 1 by default. sgn allows to transpose and invert 
-the JMatrix easily. If sgn=-1, the matrix is transposed.
 """
-JMatrix(T::Type, n::Integer) = JMatrix{T,typeof(n),Any}(n, T(1))
-JMatrix(T::Type, n::Integer, sgn) = JMatrix{T,typeof(n), Any}(n, T(sgn))
-
-Base.similar(J::JMatrix,::Type{T}) where {T} = JMatrix(T, J.n, J.sgn)
```

Comment:
Note that `similar` is documented as creating a "mutable array", so it should not return an immutable type.  Better to just omit the method here.